### ISSUE TITLE
Close a given track

### ISSUE THREAD

**Issue by scls19fr** [tumic0/GPXSee#222]
Hello,

When moving mouse on a track, it's outlined.
Maybe a contextual menu could be add (on outlined track map or elevation graph) to close outlined track (keeping others tracks opened)

Kind regards

**Comment by tumic0** [comment]
Some more fine-tuned filtering of the visible objects than the current (tracks/routes/waypoints/areas) would definitely help in some cases. The question is, how to make the filter UX/UI and keep the current simple layout without the object lists...

**Comment by scls19fr** [comment]
I think that contextual menu on track could be solution... right click on a track, contextual could should "Close this track" (but also "Zoom fit to this track" see #223)

**Comment by tumic0** [comment]
Yes, but that's only one part of the story. The other part is how you get the track back to screen (from disabled to enabled)...

**Comment by scls19fr** [comment]
"Closing a track" is not "Hide a track".
To get the track back to screen track need to be opened again.
Maybe a "Recent files" menu should be here.
If there was a "hide track" feature this would be different (but close track is also fine to me!)

**Comment by tumic0** [comment]
Many formats allow multiple tracks in one file. Shall closing a track from the file close all other tracks from the file? If not you will get a quiet inconsistent state... A file menu (like in the case of POI files) is one option that I'm thinking about, but the granularity can not be track based in this case.

**Comment by scls19fr** [comment]
oh... well... that's not so easy that I was expecting !

I don't have such a file. What is the UI behavior when mousing over one track of a multiple tracks file?
Is only the selected track outlined? Maybe outline width should be differents.... selected track with a very bold outline and tracks of the same file with a smaller outline. 
"Close tracks of this file" will close all tracks of this file (small and very bold)
With this 2 bold outline level, user closing a file will be aware that it will close several tracks at once.

But it doesn't help for hidding just one track (or several tracks) of a multiple tracks file.

Maybe a child window with 2 tabs could be considered:
- 1 tab for tracks
- 1 tab for files

Tracks tab will show a table inside with

EYE_ICON1 TIME_OF_TRACK1 FILENAME1
EYE_ICON2 TIME_OF_TRACK2 FILENAME2
EYE_ICON3 TIME_OF_TRACK3 FILENAME3

Clicking on EYE_ICON could hide or make track visible

Files tab will show a tree (QTreeView?)

filename1
  track1.1
  track1.2
  track1.3
filename2
  track2.1
  track2.2
  track2.3

Action such as hidding track could be done using contextual menu on trackX.Y items
Closing file could be done using contextual menu on filenameX items

**Comment by alranel** [comment]
@tumic0 I wouldn't be afraid of a breaking the UX if you allow the user to show/hide an object list where you can show/hide/close tracks, as well as change their color. :)
